SDK_versioned_docs/version-3.0.0/guides/auto-router/swap-and-add.md

```diff
@@ -0,0 +1,147 @@
+---
+id: swap-and-add
+title: Swap and Add Liquidity Atomically
+sidebar_position: 2
+---
+
+# Swap and Add Liquidity Atomically
+
+You can use the Auto Router to fetch optimized trade routes that result in token balances proportional to the amount needed for adding optimal liquidity to a pool position. Use the method `routeToRatio` to get quotes, gas information, and calldata for atomically swapping tokens to a particular ratio and adding the new token balances to a pool position.
+
+## Initializing the Alpha Router
+
+Follow the instructions to [import](https://docs.uniswap.org/sdk/guides/auto-router/quick-start#importing-the-package) then [initialize](https://docs.uniswap.org/sdk/guides/auto-router/quick-start#initializing-the-alpharouter) an instance of the Alpha Router
+
+## Calling `routeToRatio`
+
+The `routeToRatio` method returns all the swap and add calldata needed for submitting an atomic transaction to perform a swap and add liquidity to a position.
+
+Once you instantiate `AlphaRouter` call `routeToRatio` with the following parameters:
+
+```typescript
+{
```

Comment:
ohhhhh I finally realized the full code example comes at the end after the parameter bullet points.  I think this will trip up a lot of people. I suggest we remove this code-style parameter object.  Let's go straight into the parameter bullet points and then the full code example.

Reply:
ok fwiw, I've copied this format straight from the example using `route` so you might not think that docs page is laid out most effectively either

Reply:
ahh I see. I think the use of a table for parameter descriptions in the router doc makes a huge difference! if we use a table here then I think it's okay to keep this parameter code snippet

the bullet points just made me as a reader feel like we finished the code example and moved on to more text